src/store/balanceReducer.js

```diff
@@ -0,0 +1,17 @@
+const intialState = {
+    balance : 0
+}
+
+const balanceReducer = (state=intialState,action) => {
+switch(action.type){
+    case "DEPOSIT":
+        return {balance: state.balance + action.payload}
```

Comment:
You need to use spread operator here, like how you did in line 12. Otherwise you will end up in changing the entire state!!